lib/resource_kit/testing/action_handler_matchers.rb

```diff
@@ -21,16 +21,29 @@ def with(options, &block)
       def matches?(subject, &block)
         @handled_block ||= block
         action = subject.resources.find_action(self.action)
-        return false unless action
+        unless action
+          @failures = "expected :#{self.action} to be handled by the class."
+          return false
+        end
 
         status_code = response_stub.status || 200
-        return false unless action.handlers[status_code]
+        unless action.handlers[status_code]
+          @failures = "expected the #{status_code} status code to be handled by the class."
+          return false
+        end
 
         handled_response = action.handlers[status_code].call(response_stub)
         @handled_block.call(handled_response)
 
         true
       end
+
+      def failure_message
+        failures
+      end
+
+      private
+      attr_reader :failures
```

Comment:
If you change this to be a public reader called `failure_message` then we don't need the middleman method.